QueryBuilder/AnonymizedQueryBuilder.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the ekino/data-protection-bundle project.
+ *
+ * (c) Ekino
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Ekino\DataProtectionBundle\QueryBuilder;
+
+use Ekino\DataProtectionBundle\Annotations\AnonymizedProperty;
+use Ekino\DataProtectionBundle\Meta\AnonymizedMetadata;
+
+/**
+ * Class AnonymizedQueryBuilder.
+ * This class aims to build an anonymizing query based on an AnonymizedMetadata object.
+ *
+ * @author Benoit Mazière <benoit.maziere@ekino.com>
+ */
+final class AnonymizedQueryBuilder
+{
+    /**
+     * return string
+     */
+    public function buildQuery(AnonymizedMetadata $anonymizedMetadata): string
+    {
+        $query = '';
+
+        if ($anonymizedMetadata->getAnonymizedEntity()->isTruncateAction()) {
+            $query = $this->buildTruncateQuery($anonymizedMetadata);
+        } elseif ($anonymizedMetadata->getAnonymizedEntity()->isAnonymizeAction()) {
+            $query = $this->buildAnonymizeQuery($anonymizedMetadata);
+        }
+
+        $exceptWhereClause = $anonymizedMetadata->getAnonymizedEntity()->getExceptWhereClause() ?? '';
+
+        return $exceptWhereClause ? sprintf('%s WHERE %s', $query, $exceptWhereClause) : $query;
+    }
+
+    private function buildTruncateQuery(AnonymizedMetadata $anonymizedMetadata): string
+    {
+        return sprintf('DELETE * FROM %s', $anonymizedMetadata->getClassMetadata()->getTableName());
+    }
+
+    private function buildAnonymizeQuery(AnonymizedMetadata $anonymizedMetadata): string
+    {
+        $setters              = [];
+        $anonymizedProperties = $anonymizedMetadata->getAnonymizedProperties();
+        
+        foreach ($anonymizedProperties as $property => $anonymizedProperty) {
+            $setters[] = $this->buildPropertySetterQueryPart($anonymizedProperty);
+        }
+
+        return sprintf('UPDATE %s SET %s',
+            $anonymizedMetadata->getClassMetadata()->getTableName(), implode(', ', array_filter($setters)));
+    }
+    
+    private function buildPropertySetterQueryPart(AnonymizedProperty $anonymizedProperty): string
+    {
+        if ($anonymizedProperty->isStatic()) {
+            return sprintf('%s = "%s"', $anonymizedProperty->getColumnName(), $anonymizedProperty->getValue());
+        } elseif ($anonymizedProperty->isComposed()) {
+            $composedFieldParts = $anonymizedProperty->explodeComposedFieldValue();
+
+            return sprintf('%s = concat(concat("%s", %s), "%s")', $anonymizedProperty->getColumnName(),
```

Comment:
Yes for example I'm pretty sure the concat expression varies by platform.

Reply:
🤔 seems ok at least for mysql & posgresql. WDYT?